packages/@aws-cdk/cdk/lib/core/tag-manager.ts

```diff
@@ -94,7 +94,8 @@ export class TagManager {
   /**
    * Renders tags into the proper format based on TagType
    */
-  public renderTags(): any {
+  public renderTags(propertyTags?: any): any {
+    this.mergeFrom(propertyTags);
```

Comment:
I find this slightly scary because it will change the state of the `TagManager` class. If `renderTags()` gets called twice, do we know it does the same the 2nd time around?

I'd prefer merging on a local variable.

Reply:
agree - because of the current design it was **accidentally** idempotent, not the right thing.